File: solutions/go/leap/leap.go
// Package leap has functions for dealing with leap years.
package leap

// IsLeapYear returns if a given year is a leap year.
//
// Occurs on every year that is evenly divisible by 4
// except every year that is evenly divisible by 100
// unless the year is also evenly divisible by 400.
func IsLeapYear(year int) bool {
	if year%100 == 0 {
		return year%400 == 0
	} else {
		return year%4 == 0
	}
}

File: solutions/r/leap/leap.R
# leap returns if a given year is a leap year.
leap <- function(year) {
  # is_divisible returns if a year is evenly divisible by n.
  divisible_by <- function(n) {
    year %% n == 0
  }

  # Occurs on every year that is evenly divisible by 4
  # except every year that is evenly divisible by 100
  # unless the year is also evenly divisible by 400.
  (divisible_by(4) && !divisible_by(100)) || divisible_by(400)
}
